### ISSUE TITLE
You can exceed carry-limits by dragging into your carry slot.

### ISSUE THREAD

**Issue by johnkslg** [StrangeLoopGames/EcoIssues#21551]
![image](https://user-images.githubusercontent.com/3536496/115939009-70667080-a451-11eb-9f00-c6b168ffb915.png)

You should only be able to carry 1 dirt, and only when you have a shovel, which must be selected.
If you have a shovel that's not selected when you drag in a dirt, it should select it.

**Comment by IcedForge** [comment]
I believe this is counter productive and make any kind of Terra forming an absolute pain as well as the digging / shoveling is the absolute most disliked limitation already as is. IF this is done, then ALL carry objects should be limited to 1 including rocks, metal bars etc which would not be a "fun" experience but detrimental to the player experience.

**Comment by SlayksWood** [comment]
I agree, moreover, this is not a bug, it was specifically changed to this behavior in order to increase QOL and make this behavior not unique, because other objects have no restrictions. Pr that was approved https://github.com/StrangeLoopGames/Eco/pull/8043
